### ISSUE TITLE
Add lints IMPLICIT_SATURATING_MUL and IMPLICIT_SATURATING_DIV

### ISSUE THREAD

**Issue by alex-semenyuk** [rust-lang/rust-clippy#9606]
### What it does

Add lint similar to IMPLICIT_SATURATING_SUB and  IMPLICIT_SATURATING_ADD, but for MUL and DIV.

### Lint Name

IMPLICIT_SATURATING_MUL and IMPLICIT_SATURATING_DIV

### Category

pedantic

### Example

```rust
 let mut i: u32 = 3;
 i *= 2;

let mut i: u32 = 6;
i /= 2;
```

Could be written as:

```rust
let mut i: u32 = 3;
i = i.saturating_mul(2);

let mut i: u32 = 6;
i = i.saturating_div(2);
```

**Comment by llogiq** [comment]
I don't see anything saturating in your "bad" examples. Why should we lint assigning ops that multiply or divide?

**Comment by alex-semenyuk** [comment]
Thanks, fill the section about Advantage

**Comment by llogiq** [comment]
I might misunderstand something, but not all multiplication and division should saturate by default.

Implementing saturating multiplication manually is quite complex. Division is far easier (one only needs to consider the divide by zero case, but even then the sign needs to be applied), but still non-trivial.

So what pattern should the lint check for?
